sql/catalyst/src/main/scala/org/apache/spark/sql/util/PartitioningUtils.scala

```diff
@@ -30,14 +35,34 @@ object PartitioningUtils {
    */
   def normalizePartitionSpec[T](
       partitionSpec: Map[String, T],
-      partColNames: Seq[String],
+      partCols: StructType,
       tblName: String,
       resolver: Resolver): Map[String, T] = {
     val normalizedPartSpec = partitionSpec.toSeq.map { case (key, value) =>
-      val normalizedKey = partColNames.find(resolver(_, key)).getOrElse {
-        throw new AnalysisException(s"$key is not a valid partition column in table $tblName.")
+      val normalizedFiled = CharVarcharUtils.getRawSchema(partCols)
```

Comment:
nit: we don't need to call `getRawSchema` in the loop body. We can do `val partSchema = ...` at the beginning.

Reply:
fixed